framework/helpers/BaseHtml.php

```diff
@@ -89,6 +89,35 @@ class BaseHtml
      */
     public static $dataAttributes = ['data', 'data-ng', 'ng'];
 
+    
+    /**
+     * converts UTF-8 encoded string to app charset
+     * if string is UTF8 function returns it w/o modification
+     * @param unknown $utfString
+     * @return unknown|string
+     */
+    public static function convertToEncoding($utfString)
+    {
+    	if (strcasecmp(Yii::$app->charset, 'UTF-8') === 0) { //nothing to do
+    		return $utfString;
+    	}
+    	return mb_convert_encoding($utfString, Yii::$app->charset, 'UTF-8');
+    }
+    
+    /**
+     * converts string encoded in app charset to UTF-8
+     * if string is UTF8 function returns it w/o modification
+     * @param unknown $encString
+     * @return unknown|string
+     */
+    public static function convertToUTF($encString)
+    {
+        if (strcasecmp(Yii::$app->charset, 'UTF-8') === 0) { //nothing to do
+            return $encString;
+        }
+        return mb_convert_encoding($encString, 'UTF-8', Yii::$app->charset);
+    }
+    
```

Comment:
if file encode is different from utf8
1 - preg_match with /u expects unicode
2 - $_GET parses data from url as unicode because url is in utf8

unfortunately, i don't know how to preg_match non unicode and unicode without conversion